scripts/contracts/src/permission_management/Authorization.sol

```diff
@@ -34,16 +34,36 @@ contract Authorization is IAuthorization, ReservedAddrPublic {
         _;
     }
 
+    modifier originAdmin() {
+            require(tx.origin == allAccounts[0], "require superAdmin");
+            _;
+        }
+
     /// @notice Initialize the superAdmin's auth
     constructor(address _superAdmin) public {
-        for (uint8 i;i < builtInPermissions.length; i++)
+        for (uint8 i;i < builtInPermissions.length; i++) {
             _setAuth(_superAdmin, builtInPermissions[i]);
-
+        }
         // rootGroup: basic permissions
         _setAuth(rootGroupAddr, sendTxAddr);
         _setAuth(rootGroupAddr, createContractAddr);
     }
 
+    function updateAdmin(address _newAdmin) public originAdmin {
+        address oldAdmin = allAccounts[0];
+        if (_newAdmin == oldAdmin) {
+           return;
+        }
+        allAccounts[0] = _newAdmin;
+        allAccounts.push(oldAdmin);
+        for (uint8 i;i < builtInPermissions.length; i++) {
+            _setAuth(_newAdmin, builtInPermissions[i]);
+            // keep send tx and create contract
+            if (i != 13 && i!= 14)
```

Comment:
what meaning for `13` and `14` ? 
It is better to use a constant define?

Reply:
It is not c++